File: programming files/sum_even_fibonacci.java
// Bugs added:
// b <= 4000000 instead of b < 4000000
// b % 2 != 0 instead of b % 2 == 0
// sumEven = b instead of sumEven += b
// -EG

public class sum_even_fibonacci {
    public static void main(String[] args) {
        int a = 0, b = 1;
        int sumEven = 0;
        while (b <= 4000000) {
            if (b % 2 != 0) {
                sumEven = b;
            }
            int temp = b;
            b = a + b;
            a = temp;
        }
        System.out.println(sumEven);
    }
}

// answer should be 4613732

File: programming files/sum_even_fibonacci.py
a, b = 0, 1

# Bug added: Sum even starts at 1 instead of 0. Jonas Hemmett
sum_even = 1

while b < 4000000:
    if b % 2 == 0:
        
        # Bug added: Subtracts b instead of adding b. Jonas Hemmett
        sum_even -= b
    a, b = b, a+b

print(sum_even)

# answer should be 4613732
